dialogue-core/src/main/java/com/palantir/dialogue/core/ResponseLeakDetector.java

```diff
@@ -35,40 +32,33 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-final class LeakDetectingChannel implements Channel {
+public final class ResponseLeakDetector {
```

Comment:
thoughts on just moving this entire class to the `com.palantir.dialogue.hc4` package? this feels more internal than an intentionally exposed piece of public API

Reply:
Updated. Unfortunate we mix junit 4 and 5 here, heh